src/features/home/api/useGetOffersAndVenuesData.ts

```diff
@@ -0,0 +1,101 @@
+import { HomepageModule, isOffersModule, isVenuesModule } from 'features/home/types'
+import { useIsUserUnderage } from 'features/profile/helpers/useIsUserUnderage'
+import { SearchState } from 'features/search/types'
+import { VenueHit } from 'libs/algolia'
+import { fetchMultipleOffers } from 'libs/algolia/fetchAlgolia/fetchMultipleOffers/fetchMultipleOffers'
+import { useAdaptOffersPlaylistParameters } from 'libs/algolia/fetchAlgolia/fetchMultipleOffers/helpers/useAdaptOffersPlaylistParameters'
+import { fetchMultipleVenues } from 'libs/algolia/fetchAlgolia/fetchMultipleVenues'
+import { useTransformOfferHits, filterOfferHit } from 'libs/algolia/fetchAlgolia/transformOfferHit'
+import { useNetInfoContext } from 'libs/network/NetInfoWrapper'
+import { QueryKeys } from 'libs/queryKeys'
+import { Offer } from 'shared/offer/types'
+
+const isSearchState = (parameter: unknown): parameter is SearchState =>
+  typeof parameter === 'object' && parameter !== null
+
+export const useGetOffersAndVenuesData = (modules: HomepageModule[]) => {
+  const { position } = useHomePosition()
+  const transformHits = useTransformOfferHits()
+
+  const adaptedPlaylistParameters = useAdaptOffersPlaylistParameters()
+  const isUserUnderage = useIsUserUnderage()
+  const { user } = useAuthContext()
+  const netInfo = useNetInfoContext()
+
+  const queriesParams = modules.map((module) => {
+    if (isOffersModule(module)) {
+      const parsedParameters = module.offersModuleParameters
+        .map(adaptedPlaylistParameters)
+        .filter(isSearchState)
+      const fn = async () => {
+        const result = await fetchMultipleOffers({
+          paramsList: parsedParameters,
+          userLocation: position,
+          isUserUnderage,
+        })
+        const hits = result.hits.filter(filterOfferHit).map(transformHits)
+        return {
+          hits: uniqBy(hits, 'objectID') as Offer[],
+          nbHits: result.nbHits,
+          moduleId: module.id,
+        }
+      }
+      return { queryKeys: [QueryKeys.HOME_MODULE, module.id], queryFn: fn }
+    }
+
+    if (isVenuesModule(module)) {
+      const fn = async () => {
+        const result = await fetchMultipleVenues(module.venuesParameters, position)
+        return {
+          hits: result,
+          moduleId: module.id,
+        }
+      }
+      return { queryKeys: [QueryKeys.HOME_MODULE, module.id], queryFn: fn }
+    }
+
+    return
+  })
+
+  const queriesParamsWithoutUndefined = queriesParams.filter((q) => q !== undefined) as {
+    queryKeys: string[]
+    queryFn: () =>
+      | Promise<{
+        hits: Offer[]
+        nbHits: number
+        moduleId: string
+      }>
+      | Promise<{
+        hits: VenueHit[]
+        moduleId: string
+      }>
+  }[]
+
+  const resultList = useQueries(
+    queriesParamsWithoutUndefined.map((query) => {
+      return {
+        queryKey: query.queryKeys,
+        queryFn: query.queryFn,
+        enabled: !!netInfo.isConnected,
+      }
+    })
```

Comment:
Un avis sur cette écriture plus concise ? (si tu préfères le return explicite, ça me va aussi)
```suggestion
    queriesParamsWithoutUndefined.map((query) => ({
        queryKey: query.queryKeys,
        queryFn: query.queryFn,
        enabled: !!netInfo.isConnected,
      })
    )
```